docs/sql-migration-guide.md

```diff
@@ -36,6 +36,7 @@ license: |
   - Since Spark 3.4, Spark throws only `PartitionsAlreadyExistException` when it creates partitions but some of them exist already. In Spark 3.3 or earlier, Spark can throw either `PartitionsAlreadyExistException` or `PartitionAlreadyExistsException`.
   - Since Spark 3.4, Spark will do validation for partition spec in ALTER PARTITION to follow the behavior of `spark.sql.storeAssignmentPolicy` which may cause an exception if type conversion fails, e.g. `ALTER TABLE .. ADD PARTITION(p='a')` if column `p` is int type. To restore the legacy behavior, set `spark.sql.legacy.skipTypeValidationOnAlterPartition` to `true`.
   - Since Spark 3.4, vectorized readers are enabled by default for the nested data types (array, map and struct). To restore the legacy behavior, set `spark.sql.orc.enableNestedColumnVectorizedReader` and `spark.sql.parquet.enableNestedColumnVectorizedReader` to `false`.
+  - Since Spark 3.4, When inserting(adding/renaming/dropping) a partition, the partition field is of `String` and the partition value is not quoted, it will be treated as a field of `Numeric` by default. If you want to keep the partition field of `String`, we can set `spark.sql.legacy.keepPartitionSpecAsStringLiteral` to `true`.
```

Comment:
We already added the migration guide when introducing the breaking change, no need to add it again.